DetectorDescription/DDCMS/src/DDDetector.cc

```diff
@@ -19,6 +20,8 @@ namespace cms {
 
     m_description = &dd4hep::Detector::getInstance(tag);
     m_description->addExtension<cms::DDVectorsMap>(&m_vectors);
+    m_context = new cms::DDParsingContext(*m_description, makePayload);
```

Comment:
@cvuosalo - is it a memory leak?

Reply:
See below on line 34. In general use, when this code is not creating a DB payload, the parsing context is removed after the end of parsing. If I understand DD4hep correctly, the `removeExtension` method issues a delete on the pointer that was passed to `addExtension`.

Reply:
Please, do not expose an XML parsing context beyond an XML parser. Thanks.